cpp/src/io/orc/reader_impl.cu

```diff
@@ -724,16 +801,19 @@ void reader::impl::decode_stream_data(cudf::detail::hostdevice_2dvector<gpu::Col
                            stream);
   chunks.device_to_host(stream, true);
 
-  for (size_t i = 0; i < num_stripes; ++i) {
-    for (size_t j = 0; j < num_columns; ++j) {
-      out_buffers[j].null_count() += chunks[i][j].null_count;
+  for (size_t i = 0; i < num_columns; ++i) {
+    for (size_t j = 0; j < num_stripes; ++j) {
+      out_buffers[i].null_count() += chunks[j][i].null_count;
     }
+    // Add parent null count in case this is a child column of a struct
+    out_buffers[i].null_count() += chunks[0][i].parent_data.null_count;
   }
```

Comment:
very optional, especially given the timeline: this can be a `tabulate` with internal `accumulate`.